prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5477,6 +5540,17 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
     }
 }
 
+// TODO: is there a better way? or even place to put this? type_encoder?
+fn rust_type_name_to_vir_type(rust_type_name: &str) -> EncodingResult<vir::Type> {
```

Comment:
i'm really unhappy with this function, and still think i'd rather add the return type of lookup_pure to the `PlaceEncoding::ArrayAccess` variant, as when constructing that we still have access to the rust type, which allows simple generation of both, where now we have to do these name-based hacks..

Reply:
so this would be what i'd prefer: https://github.com/dario23/prusti-dev/compare/array-access-projection...dario23:lookup-pure-ret?expand=1

Reply:
I would also prefer to avoid this using one more field.